ymca/src/main/java/endpoints/LoggingEndpoint.java

```diff
@@ -0,0 +1,104 @@
+package endpoints;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Level;
+
+import javax.websocket.CloseReason;
+import javax.websocket.Endpoint;
+import javax.websocket.EndpointConfig;
+import javax.websocket.MessageHandler;
+import javax.websocket.Session;
+
+import org.json.JSONObject;
+
+/**
+ * Endpoint Class used to log messages and send them to the client
+ * @author Jake Lewis
+ *
+ */
+public class LoggingEndpoint extends Endpoint
+{
+
+	private static Map<String, Set<LoggingEndpoint>> endpoints = new HashMap<>();
+	private Session session;
+	private String sessionId;
+
+	@Override
+	public void onOpen(Session session, EndpointConfig config)
+	{
+		this.session = session;
+		session.addMessageHandler(new MessageHandler.Whole<String>()
+		{
+			@Override
+			public void onMessage(String text)
+			{
+				sessionId = text;
+			}
+		});
+
+		if (null == LoggingEndpoint.endpoints.get(sessionId))
+		{
+			LoggingEndpoint.endpoints.put(sessionId, new HashSet<LoggingEndpoint>());
+			LoggingEndpoint.endpoints.get(sessionId).add(this);
```

Comment:
rather than putting then getting repace with something like
```
Set mySet = new HashSet<>();
mySet.add(this);
endpoint.put(sessionId, mySet);
```